Add WCAG 2.2 (AA) for iOS (#28)
infinum/accessibility-mobile-standards: docs/guidelines/platforms/ios/guideline_understandable_ios.md

}
```

### Consistent Navigation (WCAG 3.2.3 - Level AA)

> For users is important to have a consistent navigation experience throughout the app. With that satisfied, users can easily predict where they are and where they can go next.

Reviewer: ```suggestion
For users, it is important to have a consistent navigation experience throughout the app. With that satisfied, users can easily predict where they are and where they can go next.
```

A user enters some data into a user interface element without knowing what the end result is and/or triggering an unannounced change of context in the process.

### Error Suggestion (WCAG 3.3.3 - Level AA)

To provide the best user experience, the application should suggest solutions for input errors when they are detected.

> This technique covers point *3.3.3 Error Suggestion - Level AA of the WCAG standard.*

#### ✅ Success technique(s)

> When an error is detected, the application should provide a suggestion for the user on how to correct it. This can be done by providing a hint or a description of the error, or providing a suggestion on how to correct it.

Reviewer: ```suggestion
When an error is detected, the application should provide a suggestion for the user on how to correct it. This can be done by providing a hint or a description of the error or providing a suggestion on how to correct it.
```

> This is not just an accessibility feature, but also a usability feature, as it helps users to understand what they did wrong and how to correct it. Of course, if the suggestion is presented, this should be done in accessible way (e.g. using accessibility label).
> 
> *Note: This should be avoided if the error suggestion would jeopardize the security or purpose of the content.*

Reviewer: ```suggestion
This is not just an accessibility feature, but also a usability feature, as it helps users to understand what they did wrong and how to correct it. Of course, if the suggestion is presented, this should be done in an accessible way (e.g. using an accessibility label).

*Note: This should be avoided if the error suggestion would jeopardize the security or purpose of the content.*
```

### Error Prevention (Legal, Financial, Data) (WCAG 3.3.4 - Level AA)

To prevent users from making mistakes, the application should provide a mechanism before finalizing a transaction that involves legal, financial, or data-sensitive actions.

> This technique covers point *3.3.4 Error Prevention (Legal, Financial, Data) - Level AA of the WCAG standard.*

#### ✅ Success technique(s)

To satisfy this criterion, the app should provide at least one of the following mechanisms:

- **Reversible**: The submission is reversible.
- **Checked**: Data entered by the user is checked for input errors and the user is provided with an opportunity to correct them.
- **Confirmed**: A mechanism is available for reviewing, confirming, and correcting information before finalizing the submission.

Based on the type of the application, the mechanism should be implemented in a way that is most suitable for the user and the application's purpose.

### Redundant Entry (WCAG 3.3.7 - Level A)

> Ensure that multi-step processes are easy for users by not requesting the same information multiple times in a session, as this can be challenging for those with cognitive disabilities. This approach enhances accessibility by reducing memory load and simplifying tasks.

Reviewer: ```suggestion
Ensure that multi-step processes are user-friendly by not requesting the same information multiple times in a session, as this can be challenging for those with cognitive disabilities. This approach enhances accessibility by reducing memory load and simplifying tasks.
```

- when previously entered information is no longer valid.

### Accessible Authentication (Minimum) (WCAG 3.3.8 - Level AA)

To make authentication accessible to all users, the application should not require a cognitive function test unless that step provides at least one of the following:

- **Alternative**: Another authentication method that does not rely on a cognitive function test.
- **Mechanism**: A mechanism is available to assist the user in completing the cognitive function test.
- **Object Recognition**: The cognitive function test is to recognize objects.
- **Personal Content**: The cognitive function test is to identify non-text content the user provided to the application.

> This technique covers point *3.3.8 Accessible Authentication (Minimum) - Level AA of the WCAG standard.*

#### ✅ Success technique(s)

Some of the success techniques to satisfy this criterion are:

- Support for text (copy and) paste to reduce the cognitive burden of re-typing
- Support for password entry by password managers
- Support for biometric authentication
- Support for email link authentication
- Support for OAuth authentication

#### 🚫 Failures

> To fail this criterion, users should be required to complete a cognitive function test to authenticate themselves, or enter the values in a different way than they were originally created/provided.

Reviewer: ```suggestion
To fail this criterion, users should be required to complete a cognitive function test to authenticate themselves or enter the values in a different way than they were originally created/provided.
```

Some examples:

- Write 2nd and 5th character of your password
- Write every character of the verification code in a separate field

## Other understandable guidelines

> This section contains guidelines that may not applicable for the mobile (iOS) platform, or its criteria is a not the responsibility of the mobile team. Still, take into account that those guidelines needs to be satisfied.

Reviewer: ```suggestion
This section contains guidelines that may not be applicable to the mobile (iOS) platform or its criteria is not the responsibility of the mobile team. Still, take into account that those guidelines need to be satisfied.
```
